### PR TITLE
docs: fix linting errors in architecture components guide

### PR DESCRIPTION
**Pull request by sbrshkappa**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

Issue Number: #47179 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/architecture-components.md [hunk 1]
```diff
@@ -212,8 +212,8 @@ The example template uses two built-in structural directives to add application
 
 | Directives                                  | Details |
 |:---                                         |:---     |
-| [`*ngFor`](guide/built-in-directives#ngFor) | An iterative; it tells Angular to stamp out one `<li>` per hero in the `heroes` list. |
-| [`*ngIf`](guide/built-in-directives#ngIf)   | A conditional; it includes the `HeroDetail` component only if a selected hero exists. |
+| [`*ngFor`](guide/built-in-directives#ngFor) | An iterative, it tells Angular to stamp out one `<li>` per hero in the `heroes` list. |
```
Comment (bob-watson) [inline aio/content/guide/architecture-components.md:215]: Can we just make this a complete sentence and remove the colloquialism?
```suggestion
| [`*ngFor`](guide/built-in-directives#ngFor) | An *iterative*, which tells Angular to create one `<li>` per hero in the `heroes` list. |
```

File: aio/content/guide/architecture-components.md [hunk 2]
```diff
@@ -212,8 +212,8 @@ The example template uses two built-in structural directives to add application
...
+| [`*ngIf`](guide/built-in-directives#ngIf)   | A conditional, it includes the `HeroDetail` component only if a selected hero exists. |
```
Comment (bob-watson) [inline aio/content/guide/architecture-components.md:216]: Same idea as the previous:
```suggestion
| [`*ngIf`](guide/built-in-directives#ngIf)   | A *conditional*, which includes the `HeroDetail` component only if a selected hero exists. |
```

**Review by bob-watson — CHANGES_REQUESTED** [review]
Thanks for the edits!

**Review by bob-watson — APPROVED** [review]
LGTM

Reviewed-for: global-docs-approvers

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 99c12c92a81d4019e9f3fbba2ab6d3aa478dcd96.

**Pull request closed without merging**